components/camel-aws/camel-aws2-sns/src/main/java/org/apache/camel/component/aws2/sns/Sns2Component.java

```diff
@@ -58,25 +58,29 @@ protected Endpoint createEndpoint(String uri, String remaining, Map<String, Obje
             setProperties(getCamelContext(), this, transientParameters);
         }
 
-        configuration = this.configuration != null ? this.configuration : new Sns2Configuration();
-        Sns2Endpoint endpoint = new Sns2Endpoint(uri, this, configuration);
-
         Map<String, Object> nonTransientParameters = getNonTransientParameters(parameters);
 
-        setProperties(endpoint, nonTransientParameters);
+        Sns2Configuration epConfiguration = this.configuration != null ? this.configuration.copy() : new Sns2Configuration();
 
         if (remaining.startsWith("arn:")) {
-            parseRemaining(remaining);
+            parseRemaining(epConfiguration, remaining);
         } else {
-            configuration.setTopicName(remaining);
-            LOG.debug("Created the endpoint with topic {}", configuration.getTopicName());
+            epConfiguration.setTopicName(remaining);
+            LOG.debug("Created the endpoint with topic {}", epConfiguration.getTopicName());
         }
 
-        if (!configuration.isUseDefaultCredentialsProvider() && configuration.getAmazonSNSClient() == null
-                && (configuration.getAccessKey() == null || configuration.getSecretKey() == null)) {
+        Sns2Endpoint endpoint = new Sns2Endpoint(uri, this, epConfiguration);
+        setProperties(endpoint, nonTransientParameters);
+
+        if (!epConfiguration.isUseDefaultCredentialsProvider() && epConfiguration.getAmazonSNSClient() == null
+                && (epConfiguration.getAccessKey() == null || epConfiguration.getSecretKey() == null)) {
             throw new IllegalArgumentException("AmazonSNSClient or accessKey and secretKey must be specified");
         }
 
+        if (this.configuration == null) {
+            setConfiguration(epConfiguration);
```

Comment:
Ah this seems wrong, if you set this on the component level. That should not happen.
Can you try with removing this.

Reply:
Thanks! I removed these lines and ran a test. It seems to be fine without it.